test/distributed/org/apache/cassandra/distributed/test/ReadRepairTest.java

```diff
@@ -64,14 +75,87 @@ public void readRepairTest() throws Throwable
+            for (int k = 0; k < numRows; k++)
+            {
+                assertRows(cluster.coordinator(1).execute(selectQuery, QUORUM, k), row(k, "value1", "value2"));
+            }
+
+            // verify that all the rows have meke it into node 2 because of the previous QUORUM query
+            for (int k = 0; k < numRows; k++)
+            {
+                assertRows(cluster.get(2).executeInternal(selectQuery, k), row(k, "value1", "value2"));
+            }
+        }
+    }
+
+    @Test
+    public void readRepairTimeoutTest() throws Throwable
+    {
+        final long reducedReadTimeout = 3000L;
+        try (Cluster cluster = init(builder().withNodes(3).start()))
+        {
+            cluster.forEach(i -> i.runOnInstance(() -> DatabaseDescriptor.setReadRpcTimeout(reducedReadTimeout)));
+            cluster.schemaChange("CREATE TABLE " + KEYSPACE + ".tbl (pk int, ck int, v int, PRIMARY KEY (pk, ck)) WITH read_repair='blocking'");
+            cluster.get(1).executeInternal("INSERT INTO " + KEYSPACE + ".tbl (pk, ck, v) VALUES (1, 1, 1)");
+            cluster.get(2).executeInternal("INSERT INTO " + KEYSPACE + ".tbl (pk, ck, v) VALUES (1, 1, 1)");
+            assertRows(cluster.get(3).executeInternal("SELECT * FROM " + KEYSPACE + ".tbl WHERE pk = 1"));
+            cluster.verbs(READ_REPAIR_RSP).to(1).drop();
+            final long start = System.currentTimeMillis();
+            try
+            {
+                cluster.coordinator(1).execute("SELECT * FROM " + KEYSPACE + ".tbl WHERE pk = 1", ConsistencyLevel.ALL);
+                fail("Read timeout expected but it did not occur");
+            }
+            catch (Exception ex)
+            {
+                // the containing exception class was loaded by another class loader. Comparing the message as a workaround to assert the exception
+                Assert.assertTrue(ex.getClass().toString().contains("ReadTimeoutException"));
+                long actualTimeTaken = System.currentTimeMillis() - start;
+                long magicDelayAmount = 100L; // it might not be the best way to check if the time taken is around the timeout value.
+                // Due to the delays, the actual time taken from client perspective is slighly more than the timeout value
+                Assert.assertTrue(actualTimeTaken > reducedReadTimeout);
+                // But it should not exceed too much
+                Assert.assertTrue(actualTimeTaken < reducedReadTimeout + magicDelayAmount);
+                assertRows(cluster.get(3).executeInternal("SELECT * FROM " + KEYSPACE + ".tbl WHERE pk = 1"),
+                           row(1, 1, 1)); // the partition happened when the repaired node sending back ack. The mutation should be in fact applied.
+            }
+        }
+    }
+
     @Test
     public void failingReadRepairTest() throws Throwable
     {
         try (ICluster cluster = init(builder().withNodes(3).start()))
```

Comment:
```suggestion
        try (ICluster<?> cluster = init(builder().withNodes(3).start()))
```
This is all over the place...I'd just do a quick audit.

Reply:
We could just use `Cluster` instead of `ICluster`.

Reply:
I've changed this only in the tests related to RR. We can change it in all the other tests when we are closer to merge, so we keep the PR easier to read during the review process, WDYT?